util/oidc/oidc_test.go

```diff
@@ -446,11 +447,66 @@ requestedScopes: ["oidc"]`, oidcTestServer.URL),
 
 	app.HandleCallback(w, req)
 
+	assert.Equal(t, 303, w.Code)
+	assert.NotContains(t, w.Body.String(), ErrInvalidRedirectURL.Error())
+}
+
+func Test_Login_Flow_With_PKCE(t *testing.T) {
+	var codeChallenge string
+
+	oidcTestServer := test.GetOIDCTestServer(t, func(r *http.Request) {
+		codeVerifier := r.FormValue("code_verifier")
+		assert.NotEmpty(t, codeVerifier)
+		assert.Equal(t, oauth2.S256ChallengeFromVerifier(codeVerifier), codeChallenge)
+	})
+	t.Cleanup(oidcTestServer.Close)
+
+	cdSettings := &settings.ArgoCDSettings{
+		URL: "https://example.com/argocd",
+		OIDCConfigRAW: fmt.Sprintf(`
+name: Test
+issuer: %s
+clientID: xxx
+clientSecret: yyy
+requestedScopes: ["oidc"]
+skipAudienceCheckWhenTokenHasNoAudience: true
```

Comment:
Use default value of false. This seems concerning if enablePKCEAuthentication requires skipAudienceCheckWhenTokenHasNoAudience

```suggestion
skipAudienceCheckWhenTokenHasNoAudience: false
```

Reply:
Same as above, this is because the OIDC mock server doesn’t set the audience in the token.